src/lib.rs

```diff
@@ -1,3 +1,12 @@
+#![deny(
+    rust_2018_compatibility,
+    rust_2018_idioms,
+    nonstandard_style,
+    future_incompatible,
+    unused,
+    clippy::all
+)]
```

Comment:
generally, I use `warn` for lints and then add flags in the CI builders to cargo to cause warnings to be turned into errors. I like being able to defer fixing warnings of some type while developing.

is there some reason to prefer denying all the time?